kensu/utils/kensu.py

```diff
@@ -438,46 +452,34 @@ def report_with_mapping(self):
 
                     def create_stats(schema_guid):
                         stats_df = self.real_schema_df[schema_guid]
-
-                        if isinstance(stats_df, KensuDatasourceAndSchema):
+                        stats = self.extract_stats(stats_df)
+                        if stats is None and isinstance(stats_df, KensuDatasourceAndSchema):
+                            # in this special case, the stats are reported not by kensu-py,
+                            # but by some external library called from .f_publish_stats callback
+                            # as an example - stats publishing directly from Apache Spark JVM
+                            # for pyspark's python-JVM interop jobs (where data moves between Spark & Python)
                             stats_df.f_publish_stats(lineage_run.to_guid())
-
-                        else:
-                            try:
-                                stats = self.extractors.extract_stats(stats_df)
-                            except:
-                                from kensu.requests.models import ksu_str
-                                # FIXME weird... should be fine to delete (and try,except too)
```

Comment:
removed these weird useless lines.
why would we call extract_stats twice (and second time without extra try)? cc @SammyEK